# Step 1: Load and Inspect Raw Expense Data

## Project

Personal Finance Management & Analysis System

## Objective

Load the raw financial transaction dataset and inspect its structure before cleaning.

This notebook will not modify the raw dataset.

In [3]:
import pandas as pd

print("Pandas imported successfully.")

Pandas imported successfully.


In [6]:
from pathlib import Path

PROJECT_ROOT = Path.cwd().parent.parent
RAW_FILE = PROJECT_ROOT / "data" / "raw" / "raw_expenses.csv"

print(RAW_FILE)
print("File exists:", RAW_FILE.exists())

d:\personal-finance-project\data\raw\raw_expenses.csv
File exists: True


In [7]:
df_raw = pd.read_csv(RAW_FILE)

print("Raw dataset loaded successfully.")


Raw dataset loaded successfully.


In [8]:
df_raw.head()


,Date,Account,Category,Subcategory,Note,INR,Income/Expense,Note.1,Amount,Currency
0,3/2/2022 10:11,50.0,Food,NaN,NaN,50.0,Expense,NaN,50.0,INR
1,3/2/2022 10:11,300.0,Other,NaN,NaN,300.0,Expense,NaN,300.0,INR
2,3/1/2022 19:50,78.0,Food,NaN,NaN,78.0,Expense,NaN,78.0,INR
3,3/1/2022 18:56,30.0,Transportation,NaN,NaN,30.0,Expense,NaN,30.0,INR
4,3/1/2022 18:22,67.0,Food,NaN,NaN,67.0,Expense,NaN,67.0,INR


In [9]:
print(df_raw.columns.tolist())


['Date', 'Account', 'Category', 'Subcategory', 'Note', 'INR', 'Income/Expense', 'Note.1', 'Amount', 'Currency']


In [10]:
rows, columns = df_raw.shape

print("Number of rows:", rows)
print("Number of columns:", columns)

Number of rows: 277
Number of columns: 10


In [11]:
df_raw.info()

<class 'pandas.DataFrame'>
RangeIndex: 277 entries, 0 to 276
Data columns (total 10 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Date            277 non-null    str    
 1   Account         277 non-null    float64
 2   Category        277 non-null    str    
 3   Subcategory     0 non-null      float64
 4   Note            0 non-null      float64
 5   INR             277 non-null    float64
 6   Income/Expense  277 non-null    str    
 7   Note.1          0 non-null      float64
 8   Amount          277 non-null    float64
 9   Currency        277 non-null    str    
dtypes: float64(6), str(4)
memory usage: 21.8 KB


In [12]:
missing_values = df_raw.isna().sum()

missing_values

Date                0
Account             0
Category            0
Subcategory       277
Note              277
INR                 0
Income/Expense      0
Note.1            277
Amount              0
Currency            0
dtype: int64

In [13]:
duplicate_count = df_raw.duplicated().sum()

print("Number of exact duplicate rows:", duplicate_count)

Number of exact duplicate rows: 0


In [14]:
category_counts = df_raw["Category"].value_counts()

category_counts

Category
Food                156
Other                60
Transportation       31
Apparel               7
Household             6
Allowance             6
Social Life           5
Education             1
Salary                1
Self-development      1
Beauty                1
Gift                  1
Petty cash            1
Name: count, dtype: int64

In [15]:
date_preview = pd.to_datetime(
    df_raw["Date"],
    errors="coerce"
)

print("Earliest date:", date_preview.min())
print("Latest date:", date_preview.max())
print("Unparseable dates:", date_preview.isna().sum())

Earliest date: 2021-11-21 13:30:00
Latest date: 2022-03-02 10:11:00
Unparseable dates: 0


## Initial Observations

- The dataset contains 277 rows and 10 columns.
- The transaction dates range from 2021-11-21 to 2022-03-02.
- All dates can be parsed successfully.
- The `Subcategory`, `Note`, and `Note.1` columns are completely empty.
- The `Account` column contains no missing values but requires further inspection.
- The `INR` and `Amount` columns contain numeric values.
- There are no exact duplicate rows.
- The largest categories are Food, Other, and Transportation.
- The raw dataset contains categories that must later be mapped to the project taxonomy.
- The `Salary` category represents income and should not remain an expense category.

### Next Step

In the next notebook step, we will inspect the `Account`, `INR`, `Amount`, and `Income/Expense` columns before cleaning them.